src/dms/core/EdFi.DataManagementService.Core/Security/NamedAuthorizationServiceFactory.cs

```diff
@@ -0,0 +1,59 @@
+// SPDX-License-Identifier: Apache-2.0
+// Licensed to the Ed-Fi Alliance under one or more agreements.
+// The Ed-Fi Alliance licenses this file to you under the Apache License, Version 2.0.
+// See the LICENSE and NOTICES files in the project root for more information.
+
+using System.Reflection;
+using Microsoft.Extensions.DependencyInjection;
+
+namespace EdFi.DataManagementService.Core.Security;
+
+/// <summary>
+/// Defines method for getting authorization validation/ filters handler by strategy name
+/// </summary>
+public interface IAuthorizationServiceFactory
+{
+    T? GetByName<T>(string authorizationStrategyName)
+        where T : class;
+}
```

Comment:
I still feel like this is awfully complicated for what could be a simple pattern match `switch` on strategy name -> object.